src/FloatingButton/FloatingButton.tsx

```diff
@@ -0,0 +1,169 @@
+import * as classnames from "classnames";
+import * as PropTypes from "prop-types";
+import * as React from "react";
+import * as FontAwesome from "react-fontawesome";
+
+import { Button } from "../Button/Button";
+import FlexBox from "../flex/FlexBox";
+
+import "./FloatingButton.less";
+
+const cssClass = {
+  ACTIVE: "FloatingButton--active",
+  ANIMATE: "FloatingButton--animate",
+  inactive: positionY => `FloatingButton--inactive--${positionY}`,
+
+  propStyle: prop => `FloatingButton--${prop}`,
+
+  BUTTON: "FloatingButton--button",
+  CONTAINER: "FloatingButton--container",
+  SUB_ELEMENT: "FloatingButton--subElement",
+};
+
+const propTypes = {
+  animate: PropTypes.bool,
+
+  additionalButtons: PropTypes.array,
+
+  bgColor: PropTypes.oneOf(["green"]),
+  className: PropTypes.string,
+  closeLabel: PropTypes.node,
+  label: PropTypes.node,
+
+  onClick: PropTypes.func,
+
+  offsetX: PropTypes.number,
+  offsetY: PropTypes.number,
+  positionX: PropTypes.oneOf(["left", "center", "right"]),
+  positionY: PropTypes.oneOf(["bottom", "top"]),
+  size: PropTypes.string,
+};
+
+const defaultProps = {
+  animate: true,
+  label: "+",
+  offsetX: 30,
+  offsetY: 30,
+  positionX: "right",
+  positionY: "bottom",
+};
+
+export default class FloatingButton extends React.PureComponent {
+  static propTypes = propTypes;
+  static defaultProps = defaultProps;
+  static cssClass = cssClass;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      active: false,
+    };
+  }
+
+  horizontalStyle() {
+    const { offsetX, positionX } = this.props;
+    if (positionX === "center") return {};
+    return positionX === "left"
+      ? {
+          left: `${offsetX}px`,
+        }
+      : {
+          // default to right if a bad positionX input provided
+          right: `${offsetX}px`,
```

Comment:
I know we generally use `rem` units throughout Clever for zoom-in zoom-out scaling. Rather than forcing the user to specify a pixel measure, maybe we should just let the user specify a string: `"30px"`, `"1.5rem"`. We can set up a sane `rem` default so that users don't have to worry about this for the common case

Reply:
before I commit this, I get that rem is generally better for sizing but I'm not sure it's most appropriate in this case since developers probably wouldn't want offset from edge to be tied to font size by default

Reply:
Ah yeah, I see that. I was thinking of the case that a button is positioned at the top of the window and needs to be positioned just below a header, which will likely have its height defined in `rem` units. But the more common case is probably going to be bottom positioning. And for that case, a pixel default is okay

Reply:
I realized that about rem units, deleted my comment, and used rem defaults 🤷‍♂